unstract/core/src/unstract/core/pubsub_helper.py

```diff
@@ -96,9 +91,22 @@ def log_prompt(
     @classmethod
     def publish(cls, channel_id: str, payload: dict[str, Any]) -> bool:
         channel = f"logs:{channel_id}"
+        logs_expiry = int(os.environ.get("LOGS_EXPIRATION_TIME_IN_SECOND", 3600))
         try:
-            cls.r.publish(channel, json.dumps(payload))
+            log_data = json.dumps(payload)
+            cls.r.publish(channel, log_data)
+
+            # Check if the payload type is "LOG"
+            if payload["type"] == "LOG":
+                # Extract timestamp from payload
+                timestamp = payload["timestamp"]
+
+                # Construct Redis key using channel and timestamp
+                redis_key = f"{channel}:{timestamp}"
+
+                # Store logs in Redis with expiration of 1 hour
+                cls.r.setex(redis_key, logs_expiry, log_data)
```

Comment:
Why do we have this changes here? If we do it from here, what is the purpose of `store_log` API?

Reply:
`store_log` function is used to store UI-generated logs. As an example, suppose you try to perform a task but fail in UI. For example, a user attempting to switch to the organisation they are already in.